### PR TITLE
feature:  model definition for observability analysis platform

### PR DESCRIPTION
**Pull request by liuhaoyang**
The observability analysis platform (oap) is designed to process metrics, logs and trace data, and provide unified processing, storage and query functions for observability data.
Add observability data protocol
- Metrics proto
- Trace proto
- Log proto
- Event proto
- Entity proto

### CODE REVIEW

File: oap/logs/log.proto [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import "oap/common.proto";
+
+// The logging data model 
+//   of the observability analysis platform.
+message Log {
+
+    fixed64 time_unix_nano = 1;
+
+    // log source , like contaoner_log, job_log, deploy_log, app_log .
+    string name = 2;
```
Comment (erenming) [inline oap/logs/log.proto:16]: source更好吧，name感觉不明确

File: oap/logs/log.proto [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    // The severity text (also known as log level). The original string representation as
+    // it is known at the source. [Optional].
+    string severity = 3;
+
+    // Store trace and resource pointer
+    Relation relations = 4;
+
+    // Including label, attributes and resource 
+    map<string, string> attributes = 5;
```
Comment (erenming) [inline oap/logs/log.proto:26]: 一般还需要offset: int，用来排序

Comment (liuhaoyang) [reply oap/logs/log.proto:26]: Using nanoseconds as timestamp, there is no need for offset. Before, we did not use offset

File: oap/logs/log.proto (hunk 1 above)
Comment (liuhaoyang) [reply oap/logs/log.proto:16]: `name` can be used as the source or the name of the resource, such as pod_name, container_id, etc.

File: oap/logs/log.proto (hunk 2 above)
Comment (erenming) [reply oap/logs/log.proto:26]: The precise of timestamp in the log message is not always nano seconds, which is second at most case.
When the timesamp is same, we only have to use the offset to sort

**Pull request merged**